cluster/storage/src/main/java/com/linkedin/openhouse/cluster/storage/StorageClient.java

```diff
@@ -0,0 +1,22 @@
+package com.linkedin.openhouse.cluster.storage;
+
+/**
+ * The StorageClient interface represents a client to interact with a storage system in OpenHouse.
+ * It provides a method to get the native client of the storage system.
+ *
+ * <p>Implementations of this interface should provide the specific logic for each type of storage
+ * client. For example, the {@link com.linkedin.openhouse.cluster.storage.local.LocalStorageClient}
+ * class is an implementation of this interface for local storage, and it uses an Apache Hadoop
+ * {@link org.apache.hadoop.fs.FileSystem} to interact with the local file system.
+ *
+ * @param <T> the type of the native client of the storage system
+ */
+public interface StorageClient<T> {
+
+  /**
+   * Get the native client of the storage system.
+   *
+   * @return the native client of the storage system
```

Comment:
Could it ever return a different client? Is "native" obvious and should we remove it and call it just getClient()?

Reply:
I see "native" adding value when we see the complete usage pattern:
```
Storage storage = localStorage;
storage.getClient().getNativeClient()
```
as opposed to 
```
storage.getClient().getClient()
```